lib/tasks/pipeline.js

```diff
@@ -125,35 +51,9 @@ module.exports = Task.extend({
     return pipeline.execute(context);
   },
 
-  _configuredPlugins: function(configuredPluginNames, availablePlugins) {
-    var unavailablePlugins = Object.create(null);
-    var configuredPlugins = Object.create(null);
-
-    configuredPluginNames.forEach(function(configuredPluginKey) {
-      var parts = configuredPluginKey.split(':', 2);
-      var pluginName = parts[0];
-      var configuredName = parts[1] ? parts[1] : parts[0];
-
-      if (availablePlugins[pluginName]) {
-        configuredPlugins[configuredName] = availablePlugins[pluginName];
-      } else {
-        unavailablePlugins[configuredName] = pluginName;
-      }
-    });
-
-    if (Object.keys(unavailablePlugins).length !== 0) {
-      this.ui.writeError(unavailablePlugins);
-      var error = new SilentError('plugins configuration references plugins which are not available.');
-      error.unavailablePlugins = unavailablePlugins;
-      throw error;
-    }
-
-    return configuredPlugins;
-  },
-
-  _pluginNameFromAddonName: function(addon) {
-    var pluginNameRegex = /^(ember\-cli\-deploy\-)(.*)$/;
-    return addon.name.match(pluginNameRegex)[2];
+  _plugins: function() {
+    var registry = new PluginRegistry(this.project, this.config, this.ui);
+    return registry.plugins();
```

Comment:
@achambers for when you rebase, I think the call to the validation function might go here (?)